metric/reader/reader_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package reader
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"go.opencensus.io/metric"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/metric/producer"
+)
+
+var reader *Reader
+var exporter = &metricExporter{}
+
+func TestNewReader(t *testing.T) {
```

Comment:
nit: consider adding a test when producers produce metrics but reader is already stopped.